pkg/api/annotations_test.go

```diff
@@ -693,8 +717,8 @@ func TestAPI_Annotations_AccessControl(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			setUpACL()
```

Comment:
I was just looking into how dashboard guardian is mocked, and it's a bit weird. It gets set up globally, and is done within `setUpACL` for these tests. So if the call to that function is removed here, this test will pass if it's run together with other tests, but fail on its own.

What I think we could do is:
* clean up guardian after each test. So change `setUpACL` to something like:
```
func setUpACL(t *testing.T) {
	origNewGuardian := guardian.New
	t.Cleanup(func() {
		guardian.New = origNewGuardian
	})

	viewerRole := models.ROLE_VIEWER
	editorRole := models.ROLE_EDITOR

	aclMockResp := []*models.DashboardAclInfoDTO{
		{Role: &viewerRole, Permission: models.PERMISSION_VIEW},
		{Role: &editorRole, Permission: models.PERMISSION_EDIT},
	}
	store := mockstore.NewSQLStoreMock()
	store.ExpectedDashboardAclInfoList = aclMockResp
	store.ExpectedTeamsByUser = []*models.TeamDTO{}
	guardian.InitLegacyGuardian(store)
}
```
* use mock FGAC guardian for FGAC tests, so have something like
```
func setUpFGACGuardian(t *testing.T) {
	origNewGuardian := guardian.New
	t.Cleanup(func() {
		guardian.New = origNewGuardian
	})

	guardian.MockDashboardGuardian(&guardian.FakeDashboardGuardian{CanEditValue: true})
}
```

Mock FGAC guardian doesn't add much here, and we should probably extend it in the future to look at user's permissions when evaluating access. But at least it would make it clear that a different dashboard guardian is used when FGAC is enabled.

Or if you just want to go with the fewest possible changes - keep calling `setUpACL()` from FGAC tests to make sure that they pass if they're ran separately.

Sorry for the long rant and for the messy testing setup, there's definitely a lot that can be improved here.

Reply:
Good catch! Thanks for the explanation. I will add your proposal and as you said, we should extend it in the future. For now I think we are good, the main goal of this tests is to validate FGAC annotations permissions.